Master 1.3.1 - Data Rate Fix

tohlh opened the pull request:
Greetings, I have been testing this firmware since last month. I have noticed some bugs related to data rate (DR) and decided to investigate.

This is my comment on [PR #78 of arduino-libraries/MKRWAN](https://github.com/arduino-libraries/MKRWAN/pull/78#issuecomment-632536613): 
> Every time, when I try to send the payload on the first attempt, I am always limited to **11 bytes**, regardless of what data rate I am on.
> 
> On the second attempt, the limit will always be **maximum payload size - 11 bytes**.
> 
> On the third attempt, the limit will always be **maximum payload size - 8 bytes** (which is normal).

It turns out that there are a few factors causing the above to happen:
1. When we change the DR with `AT+DR`, the DR will only change at the second `AT+SEND` command. It does not take effect on the first `AT+SEND` immediately after setting `AT+DR`.
2. When we issue `AT+JOIN` command, the DR is automatically restored to the default data rate. In my case, it is DR2. This is why I was always stuck at a maximum payload size of 11 bytes immediately after I join the network.
3. For the second attempt, my device was responding with a 3 bytes MAC command issued by my server, hence this is not a problem to fix.

The codes I added will change 2 things:
1. Make sure the data rate is changed immediately for the next `AT+SEND` whenever `AT+DR` is issued.
2. After successfully joining the network, restore the DR to the one previously configured by the user, not to the default DR.

What I am trying to achieve with my modifications is to ensure a responsive and consistent DR in control of the user. Hope my modifications above helps. Thank you.

Co-authored with: @kygoh

giulcioffi committed:
Add HYBRID for regions US915 and AU915

giulcioffi committed:
AU915 global region

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/arduino/mkrwan1300-fw?pullRequest=25) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/arduino/mkrwan1300-fw?pullRequest=25) before we can accept your contribution.<br/>**0** out of **2** committers have signed the CLA.<br/><br/>:x: giulcioffi<br/>:x: tohlh<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/arduino/mkrwan1300-fw?pullRequest=25) it.</sub>

tohlh committed:
Modified lora.c: void lora_config_tx_datarate_set(int8_t TxDataRate)

This modification allows AT+DR to change the data rate of the next AT+SEND. Previously, we could only change the data rate of the second AT+SEND.

tohlh committed:
Modified lora.c: static void MlmeConfirm(MlmeConfirm_t *mlmeConfirm)

This modification allows the data rate of the device that was set before AT+JOIN to remain even after joining the network. Previously, the data rate of the device will be restored to the default data rate for the first AT+SEND attempt after joining.

Co-Authored-By: kygoh <gohkengyew@gmail.com>

tohlh committed:
Merge branch 'master-1.3.1' into master-1.3.1
